devito/tools.py

```diff
@@ -381,3 +382,104 @@ def sweep(parameters, keys=None):
                     for v in sweep_values]
     for vals in product(*sweep_values):
         yield dict(zip(keys, vals))
+
+
+class GenericVisitor(object):
+
+    """
+    A generic visitor.
```

Comment:
... for what kind of data structure?

Reply:
Since this works for Iteration-Expression trees as well as Argument trees, I don't know how to describe it. Help?

Reply:
how about

"A generic visitor for tree data structures."

Reply:
reminder

Reply:
Do you think it might be a better idea to go back to the original docstring?

Reply:
yeah, go for it